### DISCUSSION TITLE
Looking for a new name for the project

### DISCUSSION THREAD

**Discussion by eskultety** [containerbuildsystem/cachi2#711] [Ideas]
Cachi2 is undergoing rebranding efforts as part of the ["Become an independent OSS project"](https://github.com/containerbuildsystem/cachi2/milestone/3) milestone. Part of the OSS principles is also being open about this effort and giving the community an opportunity to be part of the process and pitch some ideas. A little more background on this can be found in [2].

Here are a couple of simple rules when pitching names:
-  try coming up with a name that better describes the mission of the project - **securely prefetch build dependencies for a network isolated (i.e. hermetic) build, and record them in an accurate SBOM** 
- otherwise think of some catchy (random is fine) non-descriptive name that would not cause any confusion at all when it comes to the purpose of the project
- provide a brief reasoning behind your idea :)
- don't be afraid to like other contributor's ideas, but also please be mindful of others and don't create unnecessary noise in the discussion

Last but not least, the maintainers team would like to reserve the right to have the final call on the name that is to be ultimately picked from from all the suggestions as it's them who will be looking at it every single day going forward :) . 

Thank you!

References: 
- [1] https://github.com/containerbuildsystem/cachi2/milestone/3
- [2] https://github.com/containerbuildsystem/cachi2/issues/677

**Comment by mkosiarc** [comment]
prefetchito

**Reply by mprpic** [reply]
I'll throw in a slight alternative of this: Fetchito

**Reply by lcarva** [reply]
"fetchito" has my vote!

I think it better emphasizes the new direction of the project while giving a tip of the hat to the original name.

("cachito" was what a non-Spanish speaker came up when trying to describe something that does a little caching. The [song](https://www.youtube.com/watch?v=YCwRJe3Oo4I) was not known at the time.)

**Reply by ben-alkov** [reply]
"Fetchito"'s actually quite good

**Comment by joepvd** [comment]
After "cachito" and "cachi2", there is only one option: "CachiTool"

**Reply by mkosiarc** [reply]
I think part of the rename is to lose the part of the name that might remind people of caching, so "CachiTool" does not solve that problem

**Reply by eskultety** [reply]
Yes, it's all in written in one of the references: https://github.com/containerbuildsystem/cachi2/issues/677

**Reply by chmeliik** [reply]
On the other hand, it does make the naming even more confusing, which is quite fun

"Did he say two or tool? Or perhaps he's just pronouncing to weirdly?"

**Reply by mkosiarc** [reply]
I missed that implication, sorry @joepvd, it is quite funny in that case :D

**Comment by scoheb** [comment]
`fetching`

attractive, appealing

> The Konflux UI looks absolutely fetching!

**Comment by eskultety** [comment]
- Alpha
- Primus
- Spark

Depedency fetching precedes any hermetic application build

**Reply by adambkaplan** [reply]
"Spark" is an existing Apache project (and has a registered trademark): https://spark.apache.org/

**Reply by brunoapimentel** [reply]
Primus is a great band: https://en.wikipedia.org/wiki/Primus_(band)

(I don't think they'd care about an open source project though)

**Reply by eskultety** [reply]
> "Spark" is an existing Apache project (and has a registered trademark): https://spark.apache.org/

Oh, too bad. :( That said, the registered trademark is only on _Apache Spark_, not in general, but I agree, it would create an unnecessary google confusion - I hate that!

**Reply by eskultety** [reply]
> Primus is a great band: https://en.wikipedia.org/wiki/Primus_(band)

This would be just another disambiguation on wikipedia, wouldn't it? I'm no layer, but if this name created legal issues I can't imagine how all the disambiguations listed on wikipedia could have survived without legal charges :)

**Reply by brunoapimentel** [reply]
If we make it to the point we'd have a wikipedia page, I'd be very happy :sweat_smile: 

I don't think there'll be legal problems with the name, it means "first" in Latin.

**Comment by chmeliik** [comment]
`mobs` for SBOM backwards

Or `mops`, the czech word for this [abomination of nature](https://en.wikipedia.org/wiki/Pug)

**Reply by chmeliik** [reply]
Of course it already exists :anger: https://github.com/toadstorm/MOPS

**Reply by brunoapimentel** [reply]
Mobs is nice and short, but I'd be wary of mobs in general. They tend be very unruly.

**Reply by chmeliik** [reply]
`mobs` was mostly just going to be the lore behind `mops`, but the other project uses the same pun (even the icon is a pug) and now I'd feel like a copycat :smiling_face_with_tear:

**Comment by slimreaper35** [comment]
What do you think about a longer name ?
I don't have any concrete examples right now, but I think we could go out of the circle of short names that end with **o** ?

**Reply by eskultety** [reply]
correction - "ito" ;)

**Reply by chmeliik** [reply]
IMO short, catchy and non-descriptive is better. Choose a long, accurate name and a year later you can rebrand again :smile:

**Reply by mkosiarc** [reply]
I like the idea of keeping "ito", so we have at least some historical reference to the original

**Reply by adambkaplan** [reply]
Since this is a CLI tool/experience, long names IMO need to have a short name/acronym that we can use for the command line.

**Reply by slimreaper35** [reply]
I don't have anything against short names, but if `cachi2` was considered a bad name, does this sound better ?
- name of the organization: **fetchito**
- name of the project: **fetchito**
- CLI: **fetchito fetch-deps**

**Reply by eskultety** [reply]
> CLI: fetchito fetch-deps

I guess expressing negative preferences is also part of the process :), so I agree with @slimreaper35  and ^this is one of the reasons why I don't think `fetchito` is a great name with the other being that it creates this connotation that this tool doesn't do anything apart from fetching stuff from the internet --> not a good PR since it does a lot more - most importantly, classifying and recording each item in a good quality SBOM.

**Comment by CryptoRodeo** [comment]
Cachatouille :shallow_pan_of_food:  (_Disney pls don't sue_)

**Comment by ralphbean** [comment]
`depdep` - it pulls down deps, and is cute.

**Reply by a-ovchinnikov** [reply]
Unfortunately it is also a cutlery brand: https://www.depdep.com/

**Reply by adambkaplan** [reply]
That should be fine - completely different industry, no way someone would confuse a kitchen knife for a dev tool.

**Comment by p-rog** [comment]
If `depdep` is already used, maybe `DEPPULL`
I like `Fetchito` as well :)

**Comment by SDawley** [comment]
I also like `Fetchito`, but for the sake of throwing more options out there I would like to suggest `HermeticCLI`, solely for the reason that there is a Hermit family of hummingbirds and then the icon can be the saw-billed hermit ([wiki page](https://en.wikipedia.org/wiki/Saw-billed_hermit)).

Pros:
* small
* round

Cons:
* I couldn't come up with a decent sounding latin or -ito name that wasn't taken or produce odd results in google. though `hermetico` and `hermito` were close.

**Reply by brunoapimentel** [reply]
I actually had suggested `Hermeto` at a previous brainstorming we had. It is an actual name in Brazil (although very uncommon), and it keeps the herme- suffix that ties to the greek god Hermes (which is the origin of the word `hermetic`)

**Reply by ben-alkov** [reply]
I quite like "Hermetico" - it's short and memorable.

**Reply by SDawley** [reply]
Hermeto is also good, and easier to say out loud than 'fetchito' which has several hard consonants in a row.

**Reply by ktdreyer** [reply]
The readme says "Cachi2 itself is not a hermetic build system"

**Reply by brunoapimentel** [reply]
`hermetico` is the Portuguese word for `hermetic` :laughing:

**Reply by brunoapimentel** [reply]
> The readme says "Cachi2 itself is not a hermetic build system"

Good point, but probably the emphasis is on the "build system" part. Cachi2 is not a build system, but its goal is to enable hermetic builds as a means of having more accurate SBOMs.

**Reply by SDawley** [reply]
> `hermetico` is the Portuguese word for `hermetic` 😆

The saw-billed hermit is endemic to Brazil. I'm just saying, it's all coming together...

**Comment by dfodorRH** [comment]
_Hermesto_
It is a name "Ernesto" blended with word "hermetic". That makes (imho) a human friendly, memorable name which has an reference to purpose of the tool.

**Reply by ergonzal22** [reply]
I like it :smiley: hahaha

**Comment by kosciCZ** [comment]
I like _airlock_, since it comes across as secure and is necessary for creating a hermertically sealed environment. 

(Also there's the locking connotation towards lockfiles, which are often used here)

**Reply by adambkaplan** [reply]
:+1: on the name.

The "quick Google" due diligence turned up this company that does endpoint security. I think folks in Red Hat OSPO or Legal can check if this presents a problem: https://www.airlockdigital.com/

**Comment by adambkaplan** [comment]
Nominating `sofetch`, because this tool is "[so fetch](https://youtu.be/jjt9Qx9MBPk?si=Ax36Wx6G8RSXA6aa&t=15)."

Thinking of future CLI experience:

- `sofetch deps`
- `sofetch sbom [generate | merge ]`

and so forth...

**Comment by eskultety** [comment]
Many thanks to everyone for pitching the name ideas, the team will now evaluate internally and a new name is going to be announced in coming weeks.
